.github/workflows/master-e2e.yaml

```diff
@@ -304,6 +309,62 @@ jobs:
             # QASE_RUN_ID is empty string already
           fi
           cd tests && make start-cypress-tests
+      - name: Upgrade Rancher Manager
+        id: rancher_upgrade
+        if: ${{ inputs.rancher_upgrade != '' }}
+        env:
+          PUBLIC_DNS: ${{ needs.create-runner.outputs.public_dns }}
+          PUBLIC_DOMAIN: bc.googleusercontent.com
+          RANCHER_UPGRADE: ${{ inputs.rancher_upgrade }}
+        run: |
+          cd tests
+
+          if make e2e-upgrade-rancher-manager; then
+            # Extract Rancher Manager version
+            RANCHER_VERSION=$(kubectl get pod \
+                                --namespace cattle-system \
+                                -l app=rancher \
+                                -o jsonpath={.items[*].status.containerStatuses[*].image} 2> /dev/null || true)
+
+            # Export values
+            echo "rancher_image_version=${RANCHER_VERSION}" >> ${GITHUB_OUTPUT}
+
+          else
+            # Needed to be sure that Github Action will see the failure
+            false
+          fi
+          sleep 180
```

Comment:
not a problem but you probably know what I think about this kind of sleeps :) I'd add some sort of check that upgraded rancher is up and I'd place it into go instead.

Reply:
Yes, I completely agree with you, I am working on adding such check inside the go, if not worked then in the CI (kubectl get....)

Thanks for pointing it out. (I don't like **sleep** conditions either)